src/app/qgisapp.cpp

```diff
@@ -1664,20 +1664,10 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipBadLayers
   endProfile();
 
   // Set icon size of toolbars
-  if ( settings.contains( QStringLiteral( "/qgis/iconSize" ) ) )
-  {
-    int size = settings.value( QStringLiteral( "/qgis/iconSize" ) ).toInt();
-    if ( size < 16 )
-      size = QGIS_ICON_SIZE;
-    setIconSizes( size );
-  }
-  else
-  {
-    // first run, guess a good icon size
-    int size = chooseReasonableDefaultIconSize();
-    settings.setValue( QStringLiteral( "/qgis/iconSize" ), size );
-    setIconSizes( size );
-  }
+  int size = settings.value( QStringLiteral( "/qgis/iconSize" ), QGIS_ICON_SIZE ).toInt();
```

Comment:
Actually, I think we should also change the iconSize setting key to force all users back to the default initially. Otherwise hidpi users who have increased the icon size to fix their UI will get a massive UI on starting the new version.

Reply:
Alright, I renamed the setting key.